lib/provisioning.js

```diff
@@ -358,18 +362,45 @@ function watch(options, callback) {
 /**
  * Stops watching the specified provisioning profile directory.
  *
- * @param {String} [profileDir=~/Library/MobileDevice/Provisioning Profiles] - The path to the provisioning profile directory.
+ * @param {String} [profileDir=~/Library/Developer/Xcode/UserData/Provisioning Profiles] - The path to the provisioning profile directory.
  */
 function unwatch(profileDir) {
-	var profileDir = appc.fs.resolvePath(profileDir || defaultProfileDir);
+	var profileDirs = getExistingProvisioningProfileDirectories(profileDir);
+
+	for (const profileDir of profileDirs) {
+		if (!watchers[profileDir]) return;
```

Comment:
Need to continue and and make sure we unwatch all directories.

```suggestion
		if (!watchers[profileDir]) continue;
```

Reply:
Can you elaborate on "make sure we unwatch all directories."? I just moved the code to handle not one but multiple directories, so rest of the code is from before.

Reply:
The function is returning before it finishes looping over each directory in `profileDirs`. If this was a `forEach()`, then it would be fine, but since it's a for-of, we need to `continue` and try the next directory.

Reply:
Yeah, that one is already fixed, I meant the part afterwards. Or was that also related to that? Then it's fixed already